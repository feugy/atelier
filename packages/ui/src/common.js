import { locale, translations, getBrowserLocale } from 'svelte-intl'
import 'virtual:windi.css'
import './styles.postcss'
import fr from './locales/fr.yaml'
import en from './locales/fr.yaml'

translations.update({ fr, en })
locale.set(getBrowserLocale('fr'))
