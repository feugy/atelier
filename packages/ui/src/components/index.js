export { default as EventLogger } from './EventLogger.svelte'
export { default as Explorer } from './Explorer.svelte'
export { default as Group } from './Group.svelte'
export { default as PaneContainer } from './PaneContainer.svelte'
export { default as Toolbar } from './Toolbar.svelte'
