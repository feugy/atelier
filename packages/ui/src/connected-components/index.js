export { default as App } from './App.svelte'
export { default as ErrorDialogue } from './ErrorDialogue.svelte'
export { default as EventsPane } from './EventsPane.svelte'
export { default as PropertiesPane } from './PropertiesPane.svelte'
