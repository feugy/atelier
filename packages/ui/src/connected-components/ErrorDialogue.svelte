<script>
  import { _ } from 'svelte-intl'
  import { Dialogue } from '../components'
  import { lastError } from '../stores'
</script>

{#if $lastError}
  <Dialogue title={$_('title.error')}>
    <main>{$lastError.message}</main>
    <div slot="footer" class="text-sm">
      {$_('message.fix-and-reload')}
    </div>
  </Dialogue>
{/if}
