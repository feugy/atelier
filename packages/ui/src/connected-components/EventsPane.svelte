<script context="module">
  export const isEnabled = tool => (tool?.events || []).length > 0
</script>

<script>
  import { EventLogger } from '../components'
  import { events } from '../stores'
</script>

<EventLogger events={$events} />
