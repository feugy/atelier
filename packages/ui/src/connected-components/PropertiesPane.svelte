<script context="module">
  export const isEnabled = tool => Object.keys(tool?.props || {}).length > 0
</script>

<script>
  import { Properties } from '../components'
  import { currentTool, updateProperty } from '../stores'
</script>

<Properties properties={$currentTool?.props} on:prop-change={updateProperty} />
