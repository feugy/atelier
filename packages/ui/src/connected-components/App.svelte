<script>
  import { onMount } from 'svelte'
  import { _ } from 'svelte-intl'
  import * as EventsPane from './EventsPane.svelte'
  import * as PropertiesPane from './PropertiesPane.svelte'
  import '../common'
  import {
    Explorer,
    Frame,
    Loader,
    PaneContainer,
    Toolbar
  } from '../components'
  import {
    currentTool,
    events,
    selectTool,
    setWorkbenchFrame,
    toolsMap
  } from '../stores'

  let frame
  let viewport
  let loading = true
  $: if ($currentTool) {
    loading = false
  }
  $: console.log('loading', loading)

  onMount(() => setWorkbenchFrame(frame))
</script>

<style type="postcss">
  main {
    @apply flex-grow flex flex-col overflow-hidden;
  }

  .viewport {
    @apply flex-grow overflow-auto text-center;
    background-image: radial-gradient(
      theme('colors.secondary.light') 0.5px,
      transparent 1px
    );
    background-size: 20px 20px;
  }
</style>

<svelte:head>
  <title>{$_('title.app')}</title>
</svelte:head>

<Explorer
  toolsGroup={$toolsMap}
  current={$currentTool}
  on:select={({ detail }) => selectTool(detail)}
>
  <Toolbar {viewport} />
</Explorer>
<main>
  <div class="viewport" bind:this={viewport}>
    {#if loading}<Loader />{/if}
    <Frame bind:frame layout={$currentTool?.data?.layout} />
  </div>
  <PaneContainer
    {currentTool}
    {events}
    tabs={[
      {
        name: $_('tab.properties'),
        isEnabled: PropertiesPane.isEnabled,
        component: PropertiesPane.default
      },
      {
        name: $_('tab.events'),
        isEnabled: EventsPane.isEnabled,
        component: EventsPane.default
      }
    ]}
  />
</main>
