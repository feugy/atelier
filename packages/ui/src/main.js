import { App } from './connected-components'

new App({ target: document.body })
