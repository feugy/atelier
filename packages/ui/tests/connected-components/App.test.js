import { render, screen } from '@testing-library/svelte'
import html from 'svelte-htm'
import { translate } from '../test-utils'
import { App } from '../../src/connected-components'

describe('App connected component', () => {
  it('displays explorer, pane container and loader', () => {
    const { container } = render(html`<${App} />`)
    expect(screen.queryByRole('heading')).toHaveTextContent(
      translate('title.app')
    )
    const navs = screen.queryAllByRole('navigation')
    // container + pane containers
    expect(navs).toHaveLength(2)
    expect(container).toMatchSnapshot()
    expect(screen.getByRole('progressbar')).toBeInTheDocument()
  })
})
