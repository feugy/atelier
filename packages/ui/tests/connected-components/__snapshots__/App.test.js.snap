// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App connected component displays explorer, pane container and loader 1`] = `
<body>
  <div>
     
    <aside>
      <h1>
        <span
          class="material-icons"
        >
          handyman
        </span>
         
        <span>
          Atelier
        </span>
      </h1>
       
      <nav>
        <ul>
          <li>
            <button
              title="Modifier la couleur de fond de l'atelier"
            >
              <span
                class="material-icons"
              >
                wallpaper
              </span>
            </button>
          </li>
           
          <li>
            <button
              title="Redimensionner la taille de l'atelier"
            >
              <span
                class="material-icons"
              >
                devices
              </span>
            </button>
             
          </li>
        </ul>
      </nav>
       
      <ul />
    </aside>
     
    <main>
      <span
        class="viewport-container"
      >
        <div
          class="viewport"
        >
          <div
            class="loader"
          >
            <svg
              class="animate-spin"
              role="progressbar"
              viewBox="22 22 44 44"
            >
              <circle
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
           
          <iframe
            src="workframe.html"
            title="workframe"
          />
        </div>
      </span>
       
      <div
        class="root has-collapsed"
      >
        <nav>
          <ul>
            
            
             
            <li>
              <button
                title="Réduire/Etendre"
              >
                <span
                  class="material-icons"
                >
                  expand_less
                </span>
              </button>
            </li>
          </ul>
        </nav>
         
        <main
          class="tab-content collapsed"
        >
          
          
           
          <div
            class="disclaimer"
          >
            Cet outil n'est pas configuré pour recevoir des évènements ou pour modifier les propriétés.
          </div>
        </main>
      </div>
      
    </main>
     
    
  </div>
</body>
`;
